grype/version/constraint.go

```diff
@@ -17,6 +17,8 @@ func GetConstraint(constStr string, format Format) (Constraint, error) {
 		return newDebConstraint(constStr)
 	case RpmFormat:
 		return newRpmConstraint(constStr)
+	case PythonFormat:
+		return newFuzzyConstraint(constStr)
```

Comment:
nit: a comment highlighting why we are using the fuzzy constraint instead of a PEP440 parser would help here. The test cases added for the fuzzy constraint are great for pointing out what is supported, pairing that with a comment here that lists some specific example PEP440 features that aren't supported could help in debugging potential issues later (doesn't need to be exhaustive, as it is a very full-featured PEP)

Reply:
I added a comment with some of the oddities that this might not pick up, as well as a link to PEP440